src/basic/fileio.c

```diff
@@ -317,6 +337,22 @@ int write_string_file_ts_at(
         if (r < 0)
                 goto fail;
 
+        if (flags & WRITE_STRING_FILE_ENABLE_FS_VERITY) {
+                _cleanup_close_ int read_only_fd = -EBADF;
+
+                /* fsverity IOCTLs require an exclusive, read-only file descriptor, and fail otherwise. */
+                read_only_fd = fd_reopen(fd, O_RDONLY|O_CLOEXEC|O_NOCTTY);
+                if (read_only_fd < 0) {
+                        r = read_only_fd;
+                        goto fail;
+                }
+                f = safe_fclose(f);
+
+                r = fsverity_enable(read_only_fd, fn, /* signature= */ NULL, /* signature_size= */ 0);
+                if (r < 0 && !ERRNO_IS_NOT_SUPPORTED(r))
+                        log_debug_errno(r, "Failed to enable fs-verity on '%s', ignoring: %m", fn);
```

Comment:
i think we should debug log even in the case where the fs does not support it. it's still interesting to know when debugging...

maybe move the debug logging exclsuively into fsverity_enable()?

Reply:
The FS and its configuration (it has to be manually enabled) is up to the system builder/admin, if it's not available then we should not add noise to the log. It is well documented in the kernel docs how to enabled it, and where it's expected to work.